### DISCUSSION TITLE
updateMany on many master-detail

### DISCUSSION THREAD

**Discussion by TomasMoskala** [prisma/prisma#5221] [Q&A] [answered]
Hi, 
is there a way to write one command to update several foreign keys with the same value?
updateMany() lets me use only non-key fields
update() must be used in a cycle.

**Comment by nikolasburk** [comment, marked as answer]
Hey @TomasMoskala, thanks a lot for raising this! 

If I understand correctly, the `set` option in an `update` query might solve this issue. You can find the documentation for that [here](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#one-to-many-relations).

```ts
const user = await prisma.user.update({
  where: { email: 'alice@prisma.io' },
  data: {
    posts: {
      set: [{ id: 32 }, { id: 42 }],
    },
  },
})
```

Let me know if that helps or if you have any further questions! If that doesn't help yet, it would be great if you could share your Prisma models and a bit more details about your exact use case :)

**Reply by TomasMoskala** [reply]
Is that equal to this sql command?
```
update Post
set authorID = (select id from User where email=alice@prisma.io)
where id=32 or id=42 
```
If so then it is OK but I am also trying to use it in upsert() but the create: {} object does not have this set option. I can only use connect and create. Is this correct?

```
const user = await prisma.user.upsert({
  where: { email: 'alice@prisma.io' },
  create: {
    email: 'alice@prisma.io' 
    posts: {
      connect: [{ id: 32 }, { id: 42 }],
    },
  },
  update {
    posts: {
      set: [{ id: 32 }, { id: 42 }],
    },
  },
})
```
So connect is adding connections to detail records whereas set would replace connections (remove any existing connections that are not in the array), right?

**Reply by 2color** [reply]
> Is that equal to this sql command?
> `update Post set authorID = (select id from User where email=alice@prisma.io) where id=32 or id=42`

No. `set` will first disconnect any post that is linked to the user and then join/link the posts that are passed in.


> So connect is adding connections to detail records whereas set would replace connections (remove any existing connections that are not in the array), right?

Exactly 👍

**Reply by TomasMoskala** [reply]
OK, thanks

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
